bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -107,16 +136,80 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     ///
     /// Adding a keychain means you will be able to derive new script pubkeys under that keychain
     /// and the txout index will discover transaction outputs with those script pubkeys.
+    ///
+    /// # Panics
+    ///
+    /// This will panic if a different `descriptor` is introduced to the same `keychain`.
     pub fn add_keychain(&mut self, keychain: K, descriptor: Descriptor<DescriptorPublicKey>) {
-        // TODO: panic if already different descriptor at that keychain
-        self.keychains.insert(keychain, descriptor);
+        let old_descriptor = &*self.keychains.entry(keychain).or_insert(descriptor.clone());
+        assert_eq!(
+            &descriptor, old_descriptor,
+            "keychain already contains a different keychain"
+        );
```

Comment:
replenish lookahead?

Reply:
This is to ensure caller does not store a different descriptor under the same keychain.

Reply:
No I mean you need to replenish lookahead here :facepalm: